### ISSUE TITLE
Improve height calculation on complex layouts

### ISSUE THREAD

**Issue by kAlvaro** [nprapps/pym.js#107]
I've found incorrect height figures in a site that's using Foundation for layout. I'd modified `Child.prototype.sendHeight()` with [code taken from this StackOverflow answer](http://stackoverflow.com/a/1147768/13508):

        /**
         * Transmit the current iframe height to the parent.
         *
         * Call this directly in cases where you manually alter the height of the iframe contents.
         *
         * @memberof Child.prototype
         * @method sendHeight
         */
        this.sendHeight = function() {
            var body, html, height;

            // Get the child's height.
            body = document.getElementsByTagName('body')[0];
            html = document.getElementsByTagName('html')[0];
            height = Math.max(
                body.scrollHeight,
                body.offsetHeight,
                html.clientHeight,
                html.scrollHeight,
                html.offsetHeight
            ).toString();

            // Send the height to the parent.
            this.sendMessage('height', height);
        }.bind(this);

<sub>Feel free to just ignore this code. I understand a Pull Request is the proper way to contribute but I'm not familiar with git and I couldn't figure out how to upload my local changes.</sub>

**Comment by eads** [comment]
We'll look into this!

**Comment by eads** [comment]
Can you send an example of the height breaking?

**Comment by kAlvaro** [comment]
I'm terribly sorry... I'd swear I had found issues in my first (simple) tests but right now I cannot reproduce it in a self-contained test case and I lost those snippets time ago. As about my fix, I'd recommend that you ignore it. We found further problems with it after implementing it in our Foundation-based site: while initial height calculation was correct that way, browsing inside the iframe rendered incoherent results: higher pages would increase the iframe but lower ones would not decrease it. The ultimate cause appeared to be that inner document heigh was actually that of the current iframe, probably due to Foundation's extremely complex CSS that most likely forced the page to grow and use all space.

I'm closing this ticket. If I ever find useful information I can always open it again.
